src/code_wars/next_bigger_number.py

```diff
@@ -0,0 +1,38 @@
+# pylint: disable=invalid-name
+import math
+
+
+def find_i_of_n_with_larger_m_to_right(numbers):
+    n_with_larger_to_right = None
+    for i in range(len(numbers) - 1, 0, -1):
+        left, right = numbers[i - 1], numbers[i]
+        if left < right:
+            n_with_larger_to_right = left
+            break
+    return None if n_with_larger_to_right is None else i - 1
```

Comment:
so if numbers = [1, 2, 3]
find_i_of_n_with_larger_m_to_right(numbers) returns 1??

Reply:
```python
def find_i_of_n_with_larger_m_to_right(numbers):
    n_with_larger_to_right = None
    for i in range(len(numbers) - 1, 0, -1):
        left, right = numbers[i - 1], numbers[i]
        if left < right:
            n_with_larger_to_right = left
            break
    return None if n_with_larger_to_right is None else i - 1
```
Yes, line by line
```
line 5: numbers = [1, 2, 3]
line 6: n_with_larger_to_right = None
line 7: i = 2, n_with_larger_to_right = None
line 8: i = 2, n_with_larger_to_right = None, left = 1, right = 2
line 9: left < right is True
line 10: i = 2, n_with_larger_to_right = 1, left = 1, right = 2
line 11: and Break
line 12: return 1 because n_with_larger_to_right is not None.
```